### PR TITLE
Add contract reporter to execution-state-extract

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Added a reporter that produces a file with all the contract names and where they are deployed during execution-state-extract

An example is the file produced from the latest mainnet spork: [contract_report_1613676667.txt](https://github.com/onflow/flow-go/files/6005931/contract_report_1613676667.txt)

### CODE REVIEW

File: cmd/util/cmd/execution-state-extract/execution_state_extract.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func extractExecutionState(dir string, targetHash flow.StateCommitment, outputDi
 
 	newState, err := led.ExportCheckpointAt(targetHash,
 		[]ledger.Migration{},
-		[]ledger.Reporter{migrations.StorageReporter{Log: log, OutputDir: outputDir}},
+		[]ledger.Reporter{migrations.ContractReporter{Log: log, OutputDir: outputDir}, migrations.StorageReporter{Log: log, OutputDir: outputDir}},
```
Comment (turbolent) [inline cmd/util/cmd/execution-state-extract/execution_state_extract.go:36]: Minor: Maybe but the items/reporters on separate lines to make this more readable

**Review by turbolent — APPROVED** [review]
Nice!

File: cmd/util/ledger/migrations/contract_reporter.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	for _, p := range payload {
+		id, err := keyToRegisterID(p.Key)
+		if err != nil {
+			return err
+		}
+		if len([]byte(id.Owner)) != flow.AddressLength {
+			// not an address
+			continue
+		}
+		if id.Key != "contract_names" {
```
Comment (ramtinms) [inline cmd/util/ledger/migrations/contract_reporter.go:61]: when did we add this register? do we really have this key?

File: cmd/util/ledger/migrations/contract_reporter.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+	return nil
+}
+
+func (r ContractReporter) decodeContracts(value ledger.Value) ([]string, error) {
+	contracts := make([]string, 0)
+	if len(value) == 0 {
+		return contracts, nil
+	}
+	cborDecoder := cbor.NewDecoder(bytes.NewReader(value))
```
Comment (ramtinms) [inline cmd/util/ledger/migrations/contract_reporter.go:91]: does this works? what about those two leading zeros (contract prefix)?

File: cmd/util/ledger/migrations/contract_reporter.go (hunk 2 above)
Comment (turbolent) [reply cmd/util/ledger/migrations/contract_reporter.go:61]: It was added when multiple contracts per account were added.

Comment (turbolent) [inline cmd/util/ledger/migrations/contract_reporter.go:61]: It would be nice to (re)use a function here that knows about the key and does the decoding

Comment (janezpodhostnik) [reply cmd/util/ledger/migrations/contract_reporter.go:61]: I'll see if I can use accounts.go

File: cmd/util/ledger/migrations/contract_reporter.go (hunk 3 above)
Comment (turbolent) [reply cmd/util/ledger/migrations/contract_reporter.go:91]: What do you mean with "leading zeros (contract prefix)"?

File: cmd/util/ledger/migrations/contract_reporter.go (hunk 2 above)
Comment (janezpodhostnik) [reply cmd/util/ledger/migrations/contract_reporter.go]: everything was available, so I could reuse `accounts.go` code

**Review by turbolent — APPROVED** [review]
Very nice!

**Review by ramtinms — APPROVED** [review]

**Comment by turbolent** [comment]
@janezpodhostnik Good to go?

**Pull request merged**